main.go

```diff
@@ -0,0 +1,230 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2/textlogger"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	log    = ctrl.Log.WithName("spectro-cleanup")
+
+	// optional env vars to override default configuration
+	cleanupSeconds     int64
+	propagationPolicy  = metav1.DeletePropagationBackground
+	cleanupSecondsStr  = os.Getenv("CLEANUP_DELAY_SECONDS")
+	fileConfigPath     = os.Getenv("CLEANUP_FILE_CONFIG_PATH")
+	resourceConfigPath = os.Getenv("CLEANUP_RESOURCE_CONFIG_PATH")
+	saName             = os.Getenv("CLEANUP_SA_NAME")
+	roleName           = os.Getenv("CLEANUP_ROLE_NAME")
+	roleBindingName    = os.Getenv("CLEANUP_ROLEBINDING_NAME")
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	initConfig()
+}
+
+type DeleteObj struct {
+	schema.GroupVersionResource
+	Name      string
+	Namespace string
+}
+
+func main() {
+	ctrl.SetLogger(textlogger.NewLogger(textlogger.NewConfig()))
+	ctx := context.TODO()
+
+	config := ctrl.GetConfigOrDie()
+	client, err := ctrlclient.New(config, ctrlclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		panic(err)
+	}
+	dynamic := dynamic.NewForConfigOrDie(config)
+
+	cleanupFiles()
+	cleanupResources(ctx, client, dynamic)
+
+	os.Exit(0)
+}
+
+func initConfig() {
+	// RBAC resources used to grant the spectro cleanup Pod/DaemonSet/Job
+	// the privileges necessary to perform its cleanup
+	if saName == "" {
+		saName = "spectro-cleanup"
+	}
+	if roleName == "" {
+		roleName = "spectro-cleanup-role"
+	}
+	if roleBindingName == "" {
+		roleBindingName = "spectro-cleanup-rolebinding"
+	}
+
+	// Configuration files indicating which files and K8s resources to clean up
+	if fileConfigPath == "" {
+		fileConfigPath = "/tmp/spectro-cleanup/file-config.json"
+	}
+	if resourceConfigPath == "" {
+		resourceConfigPath = "/tmp/spectro-cleanup/resource-config.json"
+	}
+
+	// How long the spectro cleanup Pod/DaemonSet/Job will wait before self-destructing
+	if cleanupSecondsStr == "" {
+		cleanupSeconds = 30
+	} else {
+		var err error
+		cleanupSeconds, err = strconv.ParseInt(cleanupSecondsStr, 10, 64)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+// readConfig loads a configuration file from the local filesystem
+func readConfig(path, configType string) []byte {
+	path = filepath.Clean(path)
+	log.Info("Reading Spectro Cleanup config", "path", path, "configType", configType)
```

Comment:
added path sanitization to address a security concern raised by BulwarkGoSec, and passed in the correct path in the log line (initially it was passing in the global fileConfigPath)